src/jump_input.jl

```diff
@@ -1,445 +1,177 @@
-    println("========================================")
-    output = Running_MibS(name_mps, name_aux)
-
-    if displayMibS
-        println(output)
-    end
-
-    
-    if saveMibS_details
-
-        name_log = ""
-        if output_MibS == name_log
-            name_log = string(name_new, "-Dlog.txt")
-        end
-
-        open(name_log, "w") do io 
-            println(io, output)
-        end
-    end
-
-    
-    Key_Optimal = false
-    Dict_Upper_Name = Dict()
-    Dict_Lower_Name = Dict()
-    Dict_Upper_Value = Dict()
-    Dict_Lower_Value = Dict()
-
-    if occursin("Optimal solution", output)
-        Key_Optimal = true
-
-        variablesA = MOI.get(new_model, MOI.ListOfVariableIndices())
-        CntU = 0
-        CntD = 0
-        for (x, y) in MOI.enumerate(variablesA)
-            if y in lower_variables
-                Dict_Lower_Name[CntD] = y
-                Dict_Lower_Value[y] = 0
-                CntD = CntD + 1
-            else
-                Dict_Upper_Name[CntU] = y
-                Dict_Upper_Value[y] = 0
-                CntU = CntU + 1
+function _parse_output(output)
+    lines = split(output, '\n')
+    found_status = false
+    objective_value = NaN
+    upper = Dict{Int,Float64}()
+    lower = Dict{Int,Float64}()
+    for line in lines
+        if !found_status
+            if occursin("Optimal solution", line)
+                found_status = true
             end
+            continue
         end
-
-        Start_Str = findfirst("Cost =", output)[1]
-        Finish_Str = length(output)
-        output = SubString(output, Start_Str, Finish_Str)
-
-        Start_Str = 1
-        Finish_Str = findfirst("Number", output)[1]-2
-        output = SubString(output, Start_Str, Finish_Str)
-
-        
-        #println(Start_Str[1])
-        #println(output[Start_Str])
-        Result = split(output, "\n")
-        TotalObj = 0.0
-
-        for line in Result
-
-            Start_Str = findfirst("=", line)[1]
-            Start_Str = Start_Str + 1
-            Finish_Str = length(line)
-            StrNum = SubString(line, Start_Str, Finish_Str)
-            #StrNum = strip(StrNum, " ")
-            
-            Num = parse(Float64, StrNum)
-        
-            if startswith(line, "C")
-                TotalObj =  Num 
-            
-            else 
-                
-                Start_Str = findfirst("[", line)[1] + 1 
-                Finish_Str = findfirst("]", line)[1] -1
-                StrNum = SubString(line, Start_Str, Finish_Str)
-                VarIndex = parse(Int32, StrNum)
-                
-                if startswith(line, "x")
-                    VarName = Dict_Upper_Name[VarIndex]
-                    Dict_Upper_Value[VarName] = Num
-                else
-                    VarName = Dict_Lower_Name[VarIndex]
-                    Dict_Lower_Value[VarName] = Num
-                end
-            end            
+        m = match(r"([xy])\[([0-9]+)\] \= (.+)", line)
```

Comment:
Regex are magical if you haven't seen them before, but complicated to understand.

`r"([xy])\[([0-9]+)\] \= (.+)"`

* Things  in `()` get "matched" so the first thing in `()` is `m[1]`, the second `m[2]` and so on.
* `([xy])` says "match the character x or the character y
* `\[` says match exactly `[` (you have to escape it because it's a special character
* `([0-9]+)` says match a sequence of numbers from 0-9 where there is at least one character. So `0`, `11`, `124` and so on
* `]` says match exactly `]`
* ` \= ` says match exactly ` = `
* `(.+)` says match everything else until the end of the line.

Reply:
it makes life soo easy.